lib/goth/backoff.ex

```diff
@@ -0,0 +1,171 @@
+defmodule Goth.Backoff do
+  @moduledoc false
+
+  require Bitwise
+
+  alias __MODULE__
+
+  defstruct [:type, :min, :max, :state]
+
+  @typedoc """
+  Supported backoff type generators.
+  """
+  @type backoff_type :: :rand | :exp | :rand_exp
+
+  @supported_types [:rand, :exp, :rand_exp]
+
+  @typedoc """
+  ## Fields
+
+    * `type`  - Defines the backoff type to be used
+                  for min/max generation.
```

Comment:
thanks for adding docs! Please move them to the `Goth.fetch` function so the user know how to turn the knobs.